client/src/components/Comments.js

```diff
@@ -0,0 +1,171 @@
+import React, { Component } from "react"
+import PropTypes from "prop-types"
+import Loader from "./Loader"
+import Button from '@material-ui/core/Button';
+import { withStyles } from '@material-ui/core/styles';
+import IconButton from '@material-ui/core/IconButton';
+import DeleteIcon from '@material-ui/icons/Delete';
+import Card from '@material-ui/core/Card';
+import CardContent from '@material-ui/core/CardContent';
+import Input from '@material-ui/core/Input';
+import { Link } from "react-router-dom"
+
+const styles = theme => ({
+    container: {
+        display: 'flex',
+        flexWrap: 'wrap',
+        alignItems: "center",
+        flexDirection: "column"
+    },
+    textField: {
+        marginLeft: theme.spacing.unit,
+        marginRight: theme.spacing.unit,
+        width: 200,
+    },
+    dense: {
+        marginTop: 19,
+    },
+    menu: {
+        width: 200,
+    },
+    button: {
+        margin: theme.spacing.unit,
+
+    },
+    card: {
+        maxWidth: 400,
+    },
+    root: {
+        display: 'flex',
+        flexWrap: 'wrap',
+        fontFamily: "inherit",
+        margin: 10
+    },
+    cssFocused: {},
+    cssUnderline: {
+        '&:after': {
+            borderBottomColor: "#7A7A7A",
+        },
+    },
+
+});
+
+class Comments extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            message: "",
+            error: false
+        }
+        this.onChange = this.onChange.bind(this);
+        this.onSubmit = this.onSubmit.bind(this)
+    }
+
+
+    onChange = (e) => {
+        this.setState({ [e.target.name]: e.target.value });
+    }
+
+    onSubmit = e => {
+        e.preventDefault();
+        const date = new Date()
+
+        if (this.props.user) {
+            const newComment = {
+                itineraryId: this.props.itinerary._id,
+                user: this.props.user.username,
+                message: this.state.message,
+                timestamp: date
+            }
+            this.props.addComment(newComment, this.props.itinerary._id)
+            this.setState({
+                message: ""
+            })
+
+        } else {
+            this.setState({
+                error: true
+            })
+        }
+    }
+
+    // Function to delete the message when the corresponding icon is clicked
+    onDeleteClick = id => {
+        this.props.deleteComment(id)
+    }
+
+    render() {
+        const { classes } = this.props;
+        const comments = this.props.comments.comments
+        const isLoading = this.props.comments.loading
+
```

Comment:
You might find it easy to install and use something like `Prettier` to catch odd spacing issues and fix them for you automatically when developing locally.